File: GraduateStudyWebApp/GSSpa/Pages/TS/PubFun/DateTime.ts
﻿
/*-- -- -- -- -- -- -- -- -- -- --
类名:DateTime
表名:DateTime(00050565)
生成代码版本:2020.06.13.1
生成日期:2020/06/15 19:14:32
生成者:
工程名称:AGC
工程ID:0005
相关数据库:tzar.tpddns.cn,19433AGC_CS12
PrjDataBaseId:0213
模块中文名:系统参数
模块英文名:SysPara
框架-层名:纯粹类(PureClass)
编程语言:TypeScript
== == == == == == == == == == == == 
*/
/// <summary>
/// DateTime(DateTime)
/// (AutoGCLib.PureClass4TypeScript:GeneCode)
/// </summary>
export class DateTime {
    public static _CurrTabName: string = "DateTime"; //当前表名,与该类相关的表名
    public static _KeyFldName: string = "mId"; //当前表中的关键字名称,与该类相关的表中关键字名
    public mintAttributeCount = 7;
    public AttributeName = new Array("mId", "year", "month", "Day", "Hour", "Minute", "Second");
    //以下是属性变量

    private mlngmId: number = 0;    //mId
    private mintYear: number = 0;    //年
    private mintMonth: number = 0;    //月
    private mintDay: number = 0;    //日
    private mintWeek: number = 0;    //日
    private mintHour: number = 0;    //时
    private mintMinute: number = 0;    //分
    private mintSecond: number = 0;    //秒

    /// <summary>
    /// mId(说明:;字段类型:bigint;字段长度:8;是否可空:False)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get mId() {
        return this.mlngmId;
    };
    /// <summary>
    /// mId(说明:;字段类型:bigint;字段长度:8;是否可空:False)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set mId(value: number) {
        this.mlngmId = value;
    };

    /// <summary>
    /// 年(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get year() {
        return this.mintYear;
    };
    /// <summary>
    /// 年(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set year(value: number) {
        this.mintYear = value;
    };

    /// <summary>
    /// 月(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get month() {
        return this.mintMonth;
    };
    /// <summary>
    /// 月(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set month(value: number) {
        this.mintMonth = value;
    };

    /// <summary>
    /// 日(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get Day() {
        return this.mintDay;
    };
    /// <summary>
    /// 日(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set Day(value: number) {
        this.mintDay = value;
    };

    /// <summary>
    /// 日(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get week() {
        return this.mintWeek;
    };
    /// <summary>
    /// 日(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set week(value: number) {
        this.mintWeek = value;
    };

    /// <summary>
    /// 时(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get Hour() {
        return this.mintHour;
    };
    /// <summary>
    /// 时(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set Hour(value: number) {
        this.mintHour = value;
    };

    /// <summary>
    /// 分(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get Minute() {
        return this.mintMinute;
    };
    /// <summary>
    /// 分(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set Minute(value: number) {
        this.mintMinute = value;
    };

    /// <summary>
    /// 秒(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public get Second() {
        return this.mintSecond;
    };
    /// <summary>
    /// 秒(说明:;字段类型:int;字段长度:4;是否可空:True)
    /// (AutoGCLib.PureClass4TypeScript:Gen_PC_ClsProperty)
    /// </summary>
    public set Second(value: number) {
        this.mintSecond = value;
    };



    /// <summary>
    /// 常量:"mId"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_mId(): string { return "mId"; }    //mId

    /// <summary>
    /// 常量:"year"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_Year(): string { return "year"; }    //年

    /// <summary>
    /// 常量:"month"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_Month(): string { return "month"; }    //月

    /// <summary>
    /// 常量:"Day"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_Day(): string { return "Day"; }    //日

    /// <summary>
    /// 常量:"Hour"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_Hour(): string { return "Hour"; }    //时

    /// <summary>
    /// 常量:"Minute"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_Minute(): string { return "Minute"; }    //分

    /// <summary>
    /// 常量:"Second"
    /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
    /// </summary>
    public static get con_Second(): string { return "Second"; }    //秒
};

File: AGCV2/AgcPureClass/Class/SysPara/DateTime.cs

 /*-- -- -- -- -- -- -- -- -- -- --
 类名:DateTime
 表名:DateTime(00050565)
 生成代码版本:2021.07.29.1
 生成日期:2021/07/29 17:41:10
 生成者:pyf
 生成服务器IP:103.116.76.183
 工程名称:AGC
 工程ID:0005
 模块中文名:系统参数
 模块英文名:SysPara
 框架-层名:纯粹类(PureClass)
 编程语言:CSharp
 注意:1、需要公共函数层(TzPubFunction.dll)的版本:2017.12.21.01
 == == == == == == == == == == == == 
 */
using System;
using System.Text; 
using System.Collections; 
using com.taishsoft.comm_db_obj;
using com.taishsoft.common;
using com.taishsoft.datetime;
using System.Runtime.Serialization;

namespace AGC.PureClass
{
 /// <summary>
 /// DateTime(DateTime)
 /// (AutoGCLib.PureClass4CSharp:GeneCode)
 /// </summary>
[Serializable]
[DataContractAttribute]
public class DateTime 
{
protected const int mintAttributeCount = 7;
public static string[] AttributeName = new string[] {"mId", "Year", "Month", "Day", "Hour", "Minute", "Second"};
//以下是属性变量

protected long mlngmId;    //mId
protected int mintYear;    //年
protected int mintMonth;    //月
protected int mintDay;    //日
protected int mintHour;    //时
protected int mintMinute;    //分
protected int mintSecond;    //秒


 /// <summary>
 /// 常量:"mId"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_mId = "mId";    //mId

 /// <summary>
 /// 常量:"Year"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_Year = "Year";    //年

 /// <summary>
 /// 常量:"Month"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_Month = "Month";    //月

 /// <summary>
 /// 常量:"Day"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_Day = "Day";    //日

 /// <summary>
 /// 常量:"Hour"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_Hour = "Hour";    //时

 /// <summary>
 /// 常量:"Minute"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_Minute = "Minute";    //分

 /// <summary>
 /// 常量:"Second"
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:DefPropertyNameConst)
 /// </summary>
 public const string con_Second = "Second";    //秒

/// <summary>
/// 构造函数
 /// (AutoGCLib.PureClass4CSharp:Gen_PC_ClassConstructor)
/// </summary>
 public DateTime()
 {
 }

public static int AttributeCount
{
get
{
return mintAttributeCount;
}
}
/// <summary>
/// 类索引器(Indexer)函数,字符串参数
 /// (AutoGCLib.PureClass4CSharp:Gen_PC_ClsIndexer)
/// </summary>
public object this[string strAttributeName]
{
get
{
if (strAttributeName  ==  con_mId)
{
return mlngmId;
}
else if (strAttributeName  ==  con_Year)
{
return mintYear;
}
else if (strAttributeName  ==  con_Month)
{
return mintMonth;
}
else if (strAttributeName  ==  con_Day)
{
return mintDay;
}
else if (strAttributeName  ==  con_Hour)
{
return mintHour;
}
else if (strAttributeName  ==  con_Minute)
{
return mintMinute;
}
else if (strAttributeName  ==  con_Second)
{
return mintSecond;
}
return null;
}
set
{
if (strAttributeName  ==  con_mId)
{
mlngmId = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (strAttributeName  ==  con_Year)
{
mintYear = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (strAttributeName  ==  con_Month)
{
mintMonth = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (strAttributeName  ==  con_Day)
{
mintDay = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (strAttributeName  ==  con_Hour)
{
mintHour = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (strAttributeName  ==  con_Minute)
{
mintMinute = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (strAttributeName  ==  con_Second)
{
mintSecond = clsEntityBase2.TransNullToInt_S(value.ToString());
}
}
}
/// <summary>
/// 类索引器(Indexer)函数, 整型(INT)序号参数
 /// (AutoGCLib.PureClass4CSharp:Gen_PC_ClsIndexer)
/// </summary>
public object this[int intIndex]
{
get
{
if (con_mId  ==  AttributeName[intIndex])
{
return mlngmId;
}
else if (con_Year  ==  AttributeName[intIndex])
{
return mintYear;
}
else if (con_Month  ==  AttributeName[intIndex])
{
return mintMonth;
}
else if (con_Day  ==  AttributeName[intIndex])
{
return mintDay;
}
else if (con_Hour  ==  AttributeName[intIndex])
{
return mintHour;
}
else if (con_Minute  ==  AttributeName[intIndex])
{
return mintMinute;
}
else if (con_Second  ==  AttributeName[intIndex])
{
return mintSecond;
}
return null;
}
set
{
if (con_mId  ==  AttributeName[intIndex])
{
mlngmId = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (con_Year  ==  AttributeName[intIndex])
{
mintYear = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (con_Month  ==  AttributeName[intIndex])
{
mintMonth = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (con_Day  ==  AttributeName[intIndex])
{
mintDay = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (con_Hour  ==  AttributeName[intIndex])
{
mintHour = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (con_Minute  ==  AttributeName[intIndex])
{
mintMinute = clsEntityBase2.TransNullToInt_S(value.ToString());
}
else if (con_Second  ==  AttributeName[intIndex])
{
mintSecond = clsEntityBase2.TransNullToInt_S(value.ToString());
}
}
}

/// <summary>
/// mId(说明:;字段类型:bigint;字段长度:8;是否可空:False)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public long mId
{
get
{
return mlngmId;
}
set
{
 mlngmId = value;
}
}
/// <summary>
/// 年(说明:;字段类型:int;字段长度:4;是否可空:True)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public int Year
{
get
{
return mintYear;
}
set
{
 mintYear = value;
}
}
/// <summary>
/// 月(说明:;字段类型:int;字段长度:4;是否可空:True)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public int Month
{
get
{
return mintMonth;
}
set
{
 mintMonth = value;
}
}
/// <summary>
/// 日(说明:;字段类型:int;字段长度:4;是否可空:True)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public int Day
{
get
{
return mintDay;
}
set
{
 mintDay = value;
}
}
/// <summary>
/// 时(说明:;字段类型:int;字段长度:4;是否可空:True)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public int Hour
{
get
{
return mintHour;
}
set
{
 mintHour = value;
}
}
/// <summary>
/// 分(说明:;字段类型:int;字段长度:4;是否可空:True)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public int Minute
{
get
{
return mintMinute;
}
set
{
 mintMinute = value;
}
}
/// <summary>
/// 秒(说明:;字段类型:int;字段长度:4;是否可空:True)
 /// (AGC.BusinessLogicEx.clsPrjTabFldBLEx:ToString4PureClass)
/// </summary>
  [DataMember]//非常重要
 public int Second
{
get
{
return mintSecond;
}
set
{
 mintSecond = value;
}
}
}
}